vou/person.py

```diff
@@ -347,7 +346,22 @@ def will_increase_dose(
                 will_increase_dose = True
                 increase_dose_src = DoseIncreaseSource.DEALER
 
-        return {"source": increase_dose_src, "success": will_increase_dose}
+        ### TODO: Add dose type to dictionary so it doesn't change during record_dose (method of use will change each time?)
+        ##### Need to deal with will-not-increase source
+
+        ##### Need to make dose_type consistent --unless increasing dose, use the last type of dose
+
+        if (increase_dose_src == DoseIncreaseSource.WILL_NOT_INCREASE) & (len(list(self.dose_increase_record.values())) > 0):
```

Comment:
I think you're right, I'll have to double check the logic here.  It might have been written prior to incorporating that filter

Reply:
I'm still confused lol. We just defined `increase_dose_src` via the logic above, which constitutes the bulk of this function. Why would we redefine it here to the last value from the dose increase record? 

I'm guessing your initial intent for this block may have been to handle instances where `record_dose_taken` grabs the latest dose increase record to determine the source for a dose. If that source is WILL_NOT_INCREASE, it would cause an error in the parameter lookup. However, I think you've handled that case in `record_dose_taken`, where it only looks for dose increase records where the source was not WILL_NOT_INCREASE. So maybe we can get rid of this entirely?

Or maybe I'm way off base. If there's another purpose for this logic that I'm missing, let me know!

Also, unrelated comment - in this line and the one below, you can also just do `len(self.dose_increase_record)` to reduce overhead, as per my comment in `simulation.py`.

Reply:
We decided to maintain this logic for now (conditionals modified for clarity but logic unchanged.)

We may modify later if we decide we want to keep track of instances where the agent chose not to increase their dose.